isthmus/src/main/java/io/substrait/isthmus/SqlExpressionToSubstrait.java

```diff
@@ -0,0 +1,114 @@
+package io.substrait.isthmus;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.substrait.extendedexpression.ExtendedExpressionProtoConverter;
+import io.substrait.extendedexpression.ImmutableExpressionReference;
+import io.substrait.extendedexpression.ImmutableExtendedExpression;
+import io.substrait.isthmus.expression.RexExpressionConverter;
+import io.substrait.isthmus.expression.ScalarFunctionConverter;
+import io.substrait.proto.ExtendedExpression;
+import io.substrait.type.NamedStruct;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.calcite.prepare.CalciteCatalogReader;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.sql2rel.SqlToRelConverter;
+import org.apache.calcite.sql2rel.StandardConvertletTable;
+
+public class SqlExpressionToSubstrait extends SqlConverterBase {
+
+  public SqlExpressionToSubstrait() {
+    this(null);
+  }
+
+  protected SqlExpressionToSubstrait(FeatureBoard features) {
+    super(features);
+  }
+
+  private final ScalarFunctionConverter functionConverter =
+      new ScalarFunctionConverter(EXTENSION_COLLECTION.scalarFunctions(), factory);
+
+  private final RexExpressionConverter rexExpressionConverter =
+      new RexExpressionConverter(functionConverter);
+
+  /**
+   * Process to execute an SQL Expression to convert into an Extended expression protobuf message
+   *
+   * @param sqlExpression expression defined by the user
+   * @param tables of names of table needed to consider to load into memory for catalog, schema,
+   *     validate and parse sql
+   * @return extended expression protobuf message
+   * @throws SqlParseException
+   */
+  public ExtendedExpression executeSQLExpression(String sqlExpression, List<String> tables)
```

Comment:
What do you think about simply calling this method `convert`?

Strictly speaking it's not executing the SQL expression, and because it's now a method of the `SqlExpressionToSubstrait` class I think it would be redundant to call it `convertSqlExpression`.

Additionally, what do you think about changing `List<String> tables` to `List<String> createStatements` to better indicate the kinds of string we are expecting.

Reply:
Changed